index.html

```diff
@@ -3944,283 +3947,243 @@ <h4>updated</h4>
       </p>
 
       <dl>
         <dt>
-did-url
+didUrl
         </dt>
         <dd>
 A conformant <a>DID URL</a> as a single string. This is the <a>DID URL</a> to
 dereference. To dereference a <a>DID fragment</a>, the complete <a>DID URL</a>
 including the <a>DID fragment</a> MUST be used. This input is REQUIRED.
         </dd>
         <dt>
-did-url-dereferencing-input-metadata
+options
         </dt>
         <dd>
 A <a href="metadata-structure">metadata structure</a> consisting of input
 options to the <code>dereference</code> function in addition to the
-<code>did-url</code> itself. Properties defined by this specification are in
-<a href="#did-url-dereferencing-input-metadata-properties"></a>. This input is
-REQUIRED, but the structure MAY be empty.
+<code>didUrl</code> itself. Properties defined by this specification are in <a
+href="#did-url-dereferencing-options"></a>. This input is REQUIRED, but the
+structure MAY be empty.
         </dd>
       </dl>
 
       <p>
-The output variables of this function are as follows:
+The return value of the <code>dereference</code> function contains the
+following properties:
       </p>
 
       <dl>
         <dt>
-did-url-dereferencing-metadata
+dereferencingMetadata
         </dt>
         <dd>
 A <a href="metadata-structure">metadata structure</a> consisting of values
 relating to the results of the <a>DID URL dereferencing</a> process. This
 structure is REQUIRED and in the case of an error in the dereferencing process,
-this MUST NOT be empty.
-Properties defined by this specification are in
-<a href="#did-url-dereferencing-metadata-properties"></a>.
-If the dereferencing is not successful, this structure MUST contain an
-<code>error</code> property describing the error.
+this MUST NOT be empty. Properties defined by this specification are in <a
+href="#did-url-dereferencing-metadata"></a>. If the dereferencing is not
+successful, this structure MUST contain an <code>error</code> property
+describing the error.
         </dd>
         <dt>
-content-stream
+contentStream
         </dt>
         <dd>
 If the <code>dereferencing</code> function was called and successful, this MUST
-contain a <a>resource</a> corresponding to the <a>DID URL</a>.
-The <code>content-stream</code> SHOULD be a <a>DID document</a> in one of the
-conformant <a>representations</a> obtained through
-the resolution process.
-
-If the dereferencing is unsuccessful, this value MUST be empty.
+contain a <a>resource</a> corresponding to the <a>DID URL</a>. The
+<code>contentStream</code> MAY be a <a>resource</a> such as a <a href="">DID
+Document</a>,  <a href="#verification-methods">Verification Method</a>,  or <a
+href="#services">service</a> that is serializable in one of the conformant
+<a>representations</a> obtained through the resolution process
+or any other resource format that can be identified via a MIME type. If the
+dereferencing is unsuccessful, this value MUST be empty.
```

Comment:
I think the additional examples were added in the wrong spot, as written this implies that a service output is in a DID Document Representation format.

```suggestion
contain a <a>resource</a> corresponding to the <a>DID URL</a>. The
<code>contentStream</code> MAY be a <a>resource</a> such as a <a href="">DID
Document</a> that is serializable in one of the conformant
<a>representations</a>,  a <a href="#verification-methods">Verification Method</a>,  or a <a
href="#services">service</a> obtained through the resolution process. If the
dereferencing is unsuccessful, this value MUST be empty.
```

Reply:
The "or any other resource format that can be identified via a MIME type" is important, because of the whole `resource=true` angle... the thing that's returned could really be anything -- an image, a PDF file, etc.

Reply:
The change I have proposed does not limit the response at all, but we can add back the language about additional mime types in a more appropriate place. The problem I sought to fix above was that the expressed wording made it sound like a DID Document could come back as a MIME type other than a conformant representation, which we DON'T want.

```suggestion
contain a <a>resource</a> corresponding to the <a>DID URL</a>. The
<code>contentStream</code> MAY be a <a>resource</a> such as a <a href="">DID
Document</a> that is serializable in one of the conformant
<a>representations</a>,  a <a href="#verification-methods">Verification Method</a>,  a <a
href="#services">service</a>, or any other resource format that can be identified via a MIME type
and obtained through the resolution process. If the
dereferencing is unsuccessful, this value MUST be empty.
```